src/Modules/WorldRecordModule/Services/WorldRecordService.cs

```diff
@@ -0,0 +1,103 @@
+﻿using EvoSC.Common.Interfaces.Models;
+using EvoSC.Common.Interfaces.Services;
+using EvoSC.Common.Remote.EventArgsModels;
+using EvoSC.Common.Services.Attributes;
+using EvoSC.Common.Services.Models;
+using EvoSC.Modules.Official.LiveRankingModule.Models;
+using EvoSC.Modules.Official.WorldRecordModule.Interfaces;
+using EvoSC.Modules.Official.WorldRecordModule.Models;
+using Flurl;
+using Flurl.Http;
+using Microsoft.Extensions.Logging;
+
+namespace EvoSC.Modules.Official.WorldRecordModule.Services;
+
+[Service(LifeStyle = ServiceLifeStyle.Singleton)]
+public class WorldRecordService : IWorldRecordService
+{
+    private readonly ILogger<WorldRecordService> _logger;
+    private readonly IMapService _mapService;
+    private WorldRecord? _currentWorldRecord;
+
+    public WorldRecordService(ILogger<WorldRecordService> logger, IMapService mapService)
+    {
+        _logger = logger;
+        _mapService = mapService;
+    }
+
+    public async Task FetchRecord(string mapUid)
+    {
+        TMioLeaderboardResponse res = await "https://trackmania.io"
```

Comment:
While flurl is fine, I've actually found the Hawf library to work better as a repository/service type of setup in the code, and you can better group API clients together. Just for the future. In fact, there is a TMIO library already: https://www.nuget.org/packages/TrackmaniaIo.ApiClient

Reply:
Ok, I've bookmarked it, but for the test now I dont think we need it, do we?